fluent-plugin-events/test/plugin/test_in_events.rb

```diff
@@ -1,17 +1,99 @@
 require "helper"
 require "fluent/plugin/in_events.rb"
+require 'fluent/test/driver/input'
+require 'fluent/test/log'
 
 class EventsInputTest < Test::Unit::TestCase
-  setup do
-    Fluent::Test.setup
-  end
+  include SumoLogic::Kubernetes::Connector
 
-  test "failure" do
+  def setup
+    # runs before each test
+    @api_version = 'v1'
+    init_globals
+    connect_kubernetes
   end
 
-  private
+  def teardown
+    # runs after each test
+  end
 
+  def log
+    Fluent::Test::TestLogger.new
+  end
+  
   def create_driver(conf)
-    Fluent::Test::Driver::Input.new(Fluent::Plugin::EventsInput).configure(conf)
+    driver = Fluent::Test::Driver::Input.new(Fluent::Plugin::EventsInput).configure(conf)
+  end
+
+  test 'pull_resource_version correctly from eventlist' do
+    config = %([])
+    driver = create_driver(config).instance
+    driver.instance_variable_set(:@client, @client)
+
+    mock_get_events
+    resource_version = driver.pull_resource_version
+    assert_equal resource_version, '2346293'
+  end
+
+  test 'initialize_resource_version correctly for different resources' do
+    config = %([])
+    driver = create_driver(config).instance
+    driver.instance_variable_set(:@client, @client)
+
+    mock_get_config_map
+    resource = driver.initialize_resource_version
+    assert_equal resource.data['resource-version-events'], 'dummy-events-rv'
+    assert_equal resource.data['resource-version-pods'], 'dummy-pods-rv'
+    assert_equal resource.data['resource-version-services'], 'dummy-services-rv'
+  end
+
+  test 'watch_events with default type_selector' do
+    config = %([])
+    driver = create_driver(config).instance
+    driver.instance_variable_set(:@client, @client)
+    mock_watch_events
+
+    selected_events_count = get_watch_resources_count_by_type_selector(["ADDED", "MODIFIED"],
+      'api_watch_events_v1.txt')
+    driver.router.expects(:emit).times(selected_events_count).with(anything, anything, anything)
+    events = driver.start_watcher_thread
+    sleep 5
+    assert_equal events.length, 9
```

Comment:
Just wanted to clarify, so `events.length` is the number of events that the watcher thread saw, but line 58 `driver.router.expects(:emit).times(selected_events_count)` is the thing that tests that we only emitted the `ADDED` and `MODIFIED` events, right?

Reply:
That's right.